news/views.py

```diff
@@ -335,13 +335,14 @@ def form_valid(self, form):
             form.add_error(None, _("Could not register you for the event, please try again later."))
             return self.form_invalid(form)
 
-        ticket = form.save()
-        if self.request.user.is_authenticated:
-            ticket.user = self.request.user
+        if EventTicket.objects.filter(user=self.request.user, active=True, timeplace=self.timeplace, event=self.event).exists():
+            form.add_error(None, _("You are already registered for this event."))
```

Comment:
Instead of checking this and giving an error, can't we just redirect the user to their ticket if they for some reason try to access this view (both GET and POST)?

Reply:
Yes, there is no point in showing the form at all.